common/src/api/internal/nexus.rs

```diff
@@ -108,6 +109,80 @@ pub struct SledInstanceState {
 
     /// The most recent state of the sled's VMM process.
     pub vmm_state: VmmRuntimeState,
+
+    /// The current state of any in-progress migration for this instance, as
+    /// understood by this sled.
+    pub migration_state: Option<MigrationRuntimeState>,
+}
+
+/// An update from a sled regarding the state of a migration, indicating the
+/// role of the VMM whose migration state was updated.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct MigrationRuntimeState {
+    pub migration_id: Uuid,
+    pub state: MigrationState,
+    pub role: MigrationRole,
+    pub gen: Generation,
+
+    /// Timestamp for the migration state update.
+    // TODO(eliza): could this just be the VMM state timestamp?
+    pub time_updated: DateTime<Utc>,
+}
+
+/// The state of an instance's live migration.
+#[derive(
+    Clone, Copy, Debug, PartialEq, Eq, Deserialize, Serialize, JsonSchema,
+)]
+#[serde(rename_all = "snake_case")]
+pub enum MigrationState {
```

Comment:
N.B. that currently we don't have a default state for "nexus has created the migration record in the DB but the sled-agent hasn't actually reported in that the migration is in progress. I'm not *entirely* sure whether we need that or not, but we might...

Reply:
Upon further consideration, I think I'm just going to go ahead and add this now, to avoid having to do another DB migration later.